app/services/like_service.rb

```diff
@@ -5,11 +5,17 @@ def initialize(user=nil)
     @user = user
   end
 
-  def create(post_id)
+  def create_for_post(post_id)
     post = post_service.find!(post_id)
     user.like!(post)
   end
 
+  def create_for_comment(comment_id)
+    comment = comment_service.find!(comment_id)
+    post_service.find!(comment.post.id) # checks implicit for visible posts
```

Comment:
I have checked this - this function is for "like on comment", the post_id is unknown in the execution path.

Reply:
Should work this way: 
```suggestion
    post_service.find!(comment.commentable_id) # checks implicit for visible posts
```
It could be a little better because it won't instantiate `Post` object just to get it's `id`. Might save some bits of resources, in case if there are no other calls to `post` for this `comment` object.

Reply:
The commentable_id works here. Thanks for the idea.